libass/ass_render.c

```diff
@@ -2130,6 +2119,28 @@ static void reorder_text(ASS_Renderer *render_priv)
     }
 }
 
+static void apply_baseline_shear(ASS_Renderer *render_priv)
+{
+    TextInfo *text_info = &render_priv->text_info;
+    FriBidiStrIndex *cmap = ass_shaper_get_reorder_map(render_priv->shaper);
+    int32_t shear = 0;
+    double last_fay = 0;
+    for (int i = 0; i < text_info->length; i++) {
+        GlyphInfo *info = text_info->glyphs + cmap[i];
+        if (text_info->glyphs[i].linebreak || last_fay != info->fay)
+            shear = 0;
+        last_fay = info->fay;
+        if (info->skip)
+            continue;
+        while (info) {
```

Comment:
I'd use a second variable here, rather than reusing the first one and then later resetting it to its previous value. Could even be a 2-line for loop then.

Reply:
Fair point. Before I go and commit a change: I tried to keep this function similar to `reorder_text` from which it is being extracted. Do you think it’s not worth it? Or perhaps should I edit the same code in `reorder_text` while I’m at it?

Reply:
@rcombs I’ve changed this to a two-line `for` loop with a second variable. I’ve also added a commit that changes `reorder_text` to avoid similarly resetting `info` but _without_ a second variable, but I’m not 100% sure I’m happy with how it turned out, so I’d appreciate if you could glance it over and say if it looks fine to you.

Reply:
fyi: the same pattern of reusing + resetting the variable is also used eg in `preliminary_layout`, but I agree using a new var is clearer. If we want to change it, I'd say we can do that whenever we happen to modify an affected function anyway. *(which is the case for `reorder_text` in this patch series)*